x-pack/plugins/security_solution/public/common/components/url_state/initialize_redux_by_url.tsx

```diff
@@ -120,9 +151,33 @@ export const useSetInitialStateFromUrl = () => {
             });
           }
         }
+
+        if (urlKey === CONSTANTS.detailPanel) {
+          if (detailPanel != null && detailPanel.panelView && detailPanel.timelineId) {
+            if (!wasTimelineCreated) {
+              /**
+               * Within a few views, the timeline is not created before the detail panel can be added,
+               * So we create a default timeline to load the detail panel, which is then merged with
+               * The accurate props required for the timeline when createTimeline is called from that component
+               * Details on the merge are in addNewTimeline: x-pack/plugins/security_solution/public/timelines/store/timeline/helpers.ts
+               * There is one scenario where the timelineById is created beforehand, and that is the flyout.
+               * To prevent overriding any timelineById, we first check if it has been created before running the dispatch below
+               */
+              dispatch(
```

Comment:
instead of calling actions to set some value in the store, can't the flyout just read the relevant piece of url state to decide to be open or not, show a spinner or some sort of empty state if the timeline hasn't been created yet, and then show correct data once connected props change or the hook that powers it are run?

Reply:
We discussed this offline, but for posterity, yes this would work and simplify things a bit, but we would be managing state between the url and redux, and it would be best long term to just have all the state in redux. Long term goal is to improve the reducer and middleware logic to better accommodate updates like this.